packages/components/addon/components/hds/alert/index.js

```diff
@@ -11,26 +29,74 @@ export default class HdsAlertIndexComponent extends Component {
     );
   }
 
+  /**
+   * @param type
+   * @type {enum}
+   * @default page
+   * @description Determines the type of the alert.
+   */
+  get type() {
+    let { type = DEFAULT_TYPE } = this.args;
+
+    assert(
+      `@type for "Hds::Alert" must be one of the following: ${TYPES.join(
+        ', '
+      )}; received: ${type}`,
+      TYPES.includes(type)
+    );
+
+    return type;
+  }
+
+  /**
+   * @param color
+   * @type {enum}
+   * @default neutral
+   * @description Determines the color scheme for the alert.
+   */
+  get color() {
+    let { color = DEFAULT_COLOR } = this.args;
+
+    assert(
+      `@color for "Hds::Alert" must be one of the following: ${COLORS.join(
+        ', '
+      )}; received: ${color}`,
+      COLORS.includes(color)
+    );
+
+    return color;
+  }
+
   /**
    * @param icon
    * @type {string}
    * @default null
    * @description The name of the icon to be used.
    */
   get icon() {
-    return this.args.icon ?? null;
+    let { icon } = this.args;
+
+    if (icon === '') {
```

Comment:
OK, I understand: is because you want to use the empty string as a way to determine if a user doesn't want an icon at all. Here's my proposal.

We know for sure that if the `@icon` is not set (so it's `undefined`) the intent is to use the default icon associated with a color. So we can decide that the way for the user to specify they don't want an icon at all is to pass `false` to the parameter. In that case the code would look like this:

```
get icon() {
  let { icon } = this.args;

  if (icon === undefined) {
    return MAPPING_COLORS_TO_ICONS[this.color];
  } else if (icon === false) {
    return false;
  } else {
    return icon;
  }
}
```

what do you think?

Reply:
https://github.com/hashicorp/design-system/commit/88e44ddcf21ce95429704ea56f23729fafa195f0